package-res/ccc/core/cartesian/axis/abstract-cart-axis-panel.js

```diff
@@ -1553,3 +1510,281 @@ def
+    // useful constants
+    var FLAT_ANGLE = Math.PI;
+    var RIGHT_ANGLE = FLAT_ANGLE / 2;
+    var FULL_ANGLE = FLAT_ANGLE * 2;
+
+    // by default, show them all
+    layoutInfo.tickVisibilityStep = 1;
+
+    var canHide = false;
+    var canRotate = false;
+    switch(overlappedLabelsMode) {
+        case 'hide':
+            canHide = true;
+            break;
+
+        case 'rotate':
+            canRotate = true;
+            break;
+
+        case 'rotatethenhide':
+            canHide = true;
+            canRotate = true;
+            break;
+
+        default:
+            // Let them lay in the wild
+            return;
+    }
+
+    if(labelRotationDirection == null) {
+        labelRotationDirection = 1;
+    }
+
+    // slice for not changing original array
+    labelDesiredAngles = !labelDesiredAngles ? [] : labelDesiredAngles.slice().sort(function(v1, v2) {
+        return Math.abs(v1) - Math.abs(v2);
+    });
+
+    var h = layoutInfo.textHeight;
+    var w = layoutInfo.maxTextWidth;
+
+    var sMin = h * labelSpacingMin, /* parameter in em */
+        sMinH = sMin, // Between baselines
+        sMinW  = fontPxWidth + sMin; // Between sides (orthogonal to baseline)
+
+    var onBottom = axisAnchor === "bottom";
+    var onLeft = axisAnchor === "left";
+
+    var isHorizontal = onBottom || axisAnchor === "top";
+
+    if(!(w > h && distanceBetweenTicks > (isHorizontal ? sMinW : sMinH))) {
+        // we don't handle cases where the label height is bigger or equal than the maximum label width
+        // also the distanceBetweenTicks must at least be bigger than the minimum spacing
+        return;
+    }
+
+    var min_angle = 0;
+    var max_angle = FULL_ANGLE;
+    if (isHorizontal) {
+        min_angle = (distanceBetweenTicks - sMinW) < w ? pvc.normAngle(Math.asin(h / (distanceBetweenTicks - sMinH))) : 0;
+    } else {
+        max_angle = pvc.normAngle(Math.acos(h / (distanceBetweenTicks - sMinH)));
+    }
+
+    // The angle that the text makes to the x axis (clockwise, y points downwards)
+    // it is normalized to the interval [0, 2π[
+    var a;
+    if(canRotate) {
+        if(labelDesiredAngles.length > 0) {
+            // choose the first angle above the minimum non-overlapping angle
+            // if none, chooses to the last desired angle
+            for (var i = 0; i !== labelDesiredAngles.length; ++i) {
+                a = pvc.normAngle(labelDesiredAngles[i] * labelRotationDirection);
+
+                // angular distance to the closest axis
+                var abs_angle = a > FLAT_ANGLE ? Math.abs(a - FULL_ANGLE) : a;
+                var angle_to_axis = abs_angle < RIGHT_ANGLE ? abs_angle : abs_angle - FLAT_ANGLE;
+                if(angle_to_axis >= min_angle) {
+                    // no need for hiding if between min and max angle, it will never overlap
+                    canHide = canHide && (isNaN(max_angle) || angle_to_axis > max_angle);
```

Comment:
By what I could understand, the `NaN` in min_angle and max_angle can only occur due to the arguments of `Math.asin` or `Math.acos` being out of range (-1, 1). What situations do these correspond to? Do they actually happen?
It feels strange to me that we don't guard against `NaN` occurrence closer to the "offending" calculations.
So, if in this line we guard against `max_angle` being `NaN`, why don't we do the same for the previous line, when using `min_angle`? Or are we relying on that comparing to `NaN` never returns `true`? In that case, why bother determining the value of `angle_to_axis`?

Mostly, I'd like that the comments would give some clue of what "use case" NaN corresponds to, if any. That would provide meaning for the following `isNaN` tests.